pallets/did/src/lib.rs

```diff
@@ -291,6 +291,31 @@ where
 	}
 }
 
+/// A DID deletion request. It contains the following values:
+/// * The DID identifier being deleted
+/// As replaying a deletion request does not change the state of the system,
+/// the deletion request does not have a tx counter.
```

Comment:
Can I recreate a did with the same identifier?

Reply:
Potentially, yes. But it will likely have a different set of keys.

Reply:
If I accidentally deleted my DID and recreate it with the same keys because I'm lazy, everyone can delete it again?

If it's possible, I would add a nonce to prevent it. Just to make it waterproof.

Reply:
In this case the counter will be reset to 0, and anyone could still re-delete the DID. To prevent that, we should allow users to specify a custom tx counter upon DID creation, or otherwise we could not completely delete the DID from the chain but just mark it as unusable, but then we are adding more complex logic there.

Reply:
He could do a batch TX that contains a DID-Create and DID-Update operation. In the update operation he then can update the nonce to a custom value. I think this is fine. 👍🏻

Reply:
Yeah, I guess having a tx counter makes sense then ✌🏻